tools/cp-caps/rcp_caps_test.py

```diff
@@ -268,9 +281,216 @@ def test_link_metrics(self):
+        self.__dut.factory_reset()
+        active_dataset = self.__get_default_dataset()
+
+        frames = [
+            self.Frame(
+                name='mIsSecurityProcessed=True',
+                tx_frame='09ec00dddd102030405060708001020304050607080d0000000001db622c220fde64408db9128c93d50000',
+                dst_address='8070605040302010',
+                is_security_processed=True),
+            self.Frame(
+                name='mIsSecurityProcessed=False',
+                tx_frame='09ec00dddd102030405060708001020304050607080d000000000000010203040506070809000000000000',
+                expect_rx_frame=
+                '09ec00dddd102030405060708001020304050607080d0000000001db622c220fde64408db9128c93d50000',
+                is_security_processed=False,
+                dst_address='8070605040302010',
+                active_dataset=active_dataset,
+                src_ext_address='0807060504030201'),
+        ]
+
+        for frame in frames:
+            ret = self.__test_send_formatted_frame(self.__dut, self.__ref, frame)
+            self.__output_format_bool(frame.name, ret)
+
+    def __test_radio_frame_tx_info_tx_delay(self):
+        self.__dut.factory_reset()
+        self.__ref.factory_reset()
+
+        # Enable the IPv6 interface to force the host and the RCP to synchronize the radio time.
+        self.__dut.set_dataset_bytes('active', self.__get_default_dataset())
+        self.__dut.ifconfig_up()
+        self.__dut.wait(0.5)
+        self.__dut.ifconfig_down()
+
+        self.__dut.diag_start()
+        self.__ref.diag_start()
+
+        channel = 11
+        packets = 1
+        dut_tx_delay_sec = 0.5
+        dut_tx_delay_us = int(dut_tx_delay_sec * 1000000)
+        ref_rx_delay_sec = dut_tx_delay_sec / 2
+        ref_address = 'dead00beefcafe01'
+        dut_tx_frame = '01ec00dddd01fecaefbe00adde02fecaefbe00adde000102030405060708090000'
+
+        self.__dut.diag_set_channel(channel)
+        self.__ref.diag_set_channel(channel)
+        self.__ref.diag_radio_receive()
+        self.__ref.diag_set_radio_receive_filter_dest_mac_address(ref_address)
+        self.__ref.diag_enable_radio_receive_filter()
+
+        self.__dut.diag_frame(dut_tx_frame, tx_delay=dut_tx_delay_us)
+        self.__dut.diag_send(packets, is_async=True)
+
+        self.__ref.wait(ref_rx_delay_sec)
+        stats = self.__ref.diag_get_stats()
+        ret = stats['received_packets'] == 0
+
+        if ret is True:
+            self.__ref.wait(ref_rx_delay_sec)
+            stats = self.__ref.diag_get_stats()
+            ret = stats['received_packets'] == 1
+
+        self.__output_format_bool(f'mTxDelayBaseTime=now,mTxDelay={dut_tx_delay_us}', ret)
+
+        self.__ref.diag_stop()
+        self.__dut.diag_stop()
+
+    def __test_radio_frame_tx_info_rx_channel_after_tx_done(self):
+        self.__dut.factory_reset()
+        self.__ref.factory_reset()
+
+        self.__dut.diag_start()
+        self.__ref.diag_start()
```

Comment:
There're setup/teardown steps in test cases. Probably you can leverage unit testing framework or make a decorator. I'm fine with not refactoring it in this PR.

Reply:
Will do it in another commit later.